lms/djangoapps/bulk_email/tests/test_email.py

```diff
@@ -268,6 +384,41 @@ def test_unicode_students_send_to_all(self):
             [self.instructor.email] + [s.email for s in self.staff] + [s.email for s in self.students]
         )
 
+    def test_long_course_display_name(self):
+        """
+        This test tests that courses with exorbitantly large display names
+        can still send emails, since it appears that 320 appears to be the
+        character length limit of from emails for Amazon SES.
+        """
+        test_email = {
+            'action': 'Send email',
+            'send_to': 'myself',
+            'subject': 'test subject for self',
+            'message': 'test message for self'
+        }
+
+        # make very long display_name for course
+        long_name = u"x" * 321
+        course = CourseFactory.create(display_name=long_name)
+        instructor = InstructorFactory(course_key=course.id)
+
+        self.login_as_user(instructor)
+        send_mail_url = reverse('send_email', kwargs={'course_id': unicode(course.id)})
+        response = self.client.post(send_mail_url, test_email)
+        self.assertTrue(json.loads(response.content)['success'])
+
+        self.assertEqual(len(mail.outbox), 1)
+        from_email = mail.outbox[0].from_email
+
+        self.assertEqual(len(from_email), 318)
+        self.assertIn(u"...", from_email)
```

Comment:
I'd like you to test that the truncation happens in the right place. I would test that from address is exactly what you expect.